platform/src/main/java/org/stellar/anchor/platform/payment/observer/circle/CirclePaymentObserverService.java

```diff
@@ -201,8 +200,15 @@ public void handleTransferNotification(CircleNotification circleNotification)
     }
 
     if (isWalletTracked(destination)) {
-      final ObservedPayment finalObservedPayment = observedPayment;
-      observers.forEach(observer -> observer.onReceived(finalObservedPayment));
+      for (PaymentListener listener : observers) {
+        try {
+          listener.onReceived(observedPayment);
+        } catch (EventPublishException ex) {
+          Log.errorEx("Failed to send event to observer.", ex);
+          Log.info("Restarting the Stellar observer.");
```

Comment:
If this is not where the observer is restarted, it's better not to log it here.

Reply:
Updated in https://github.com/stellar/java-stellar-anchor-sdk/pull/607/commits/247705aab4bb5e0d048fba7d6ae2d9c768103249.